.github/workflows/tools.yml

```diff
@@ -0,0 +1,29 @@
+name: "tools update"
+on:
+  schedule:
+    # Run once a week at 00:05 AM UTC on Saturday.
+    - cron: '5 0 * * 6'
+
+  workflow_dispatch:
+
+jobs:
+  tools_update:
+    if: github.repository == 'nodejs/node'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: tools/update-eslint.sh
+      - run: tools/update-babel-eslint.sh
+      - run: |
+          (cd tools/lint-md && rm -rf package-lock.json node_modules && npm install --ignore-scripts)
+          make lint-md-rollup
```

Comment:
I'm not sure I like this. I would prefer to have separate commits, one for each tool, with descriptive messages like "tools: update ESLint to x.x.x" as we have now.

Reply:
I agree with lpinca, and by the way, with a `matrix` it is done quite cleanly.
```yml
   strategy:
     matrix:
       include:
         - id: eslint
           run: tools/update-eslint.sh
         ...
   steps:
     ...
     - run: |
         ${{ matrix.run }}
         echo "NEW_VERSION=..." >> $GITHUB_ENV
     - uses: gr2m/create-or-update-pull-request-action@v1  # Create a PR or update the Action's existing PR
       ...
       with:
         body: "This is an automated update of ${{ matrix.id }} to ${{ env.NEW_VERSION }}."
         branch: "actions/tools-update-${{ matrix.id }}"  # Custom branch *just* for this Action.
         commit-message: "tools: update ${{ matrix.id }} to ${{ env.NEW_VERSION }}"
         title: "tools: update ${{ matrix.id }} to ${{ env.NEW_VERSION }}"
         ...
```

Reply:
@Mesteery I think that will do it as separate commits, but also separate pull requests. That will work, but I wonder if there's a straightforward way to have them be separate commits but a single pull request. Otherwise, especially as we add tools, the number of pull requests could get annoying. One pull request a week covering the tools directory but splitting each tool into its own commit (so individual things can be removed or reverted if they are problematic) would be my preference.

Reply:
Oh yes, indeed, I misread it, I thought he was suggesting to do it as separate pull requests.

> I wonder if there's a straightforward way to have them be separate commits but a single pull request.

It's possible with [`peter-evans/create-pull-request`](https://github.com/peter-evans/create-pull-request#controlling-commits), but IMO it is not very pratical.

Reply:
All right, we'll just go with separate pull requests for now and see how it goes. Thanks!

Reply:
OK, all set up (but not yet tested) with a matrix. (Thanks, @Mesteery!) I'm still pondering whether or how to get versions into the commit log. (This doesn't update only when the tool itself has a version bump. It updates dependencies even if the tool itself is the same version. So it's not as simple as, for example, checking to see what version of ESLint is installed.) Hopefully that's not a dealbreaker.

Reply:
It's good to keep our tools up to date, but isn's once a week a bit too much? I'm pretty sure we will get at least one pull request every time this is run, because the ESLint and Babel dependency trees often have updates.

Reply:
> It's good to keep our tools up to date, but isn's once a week a bit too much? I'm pretty sure we will get at least one pull request every time this is run, because the ESLint and Babel dependency trees often have updates.

Yes, it will almost certainly update once a week. (The remark dependencies are active too in `lint-md` too.) Would you be more comfortable with once every two weeks instead or something like that?

A better option might be to run `npm outdated` and only update if a top-level dependency is out of date. My hope was to keep all the transitive dependencies up to date, as that has mattered for remark linting. But we can not worry about that for now. (Maybe we can talk about auto-landing these kinds of PRs if all tests pass to minimize the maintenance burden and then running updates frequently for transitive dependencies is not a big deal.)

Reply:
(Also, running `npm outdated` first allows us to get the version number to include in the commit messages and pR text.)

Reply:
Still needs to be tested but this will hopefully now only open a pull request:

* if eslint has been updated
* if @babel/eslint-parser has been updated
* if a direct dependency of lint-md has been updated

It will open one pull request for each bullet point above, so up to 3 are possible, but will probably usually be zero or one.